sway-lsp/src/capabilities/hover.rs

```diff
@@ -15,27 +18,51 @@ use sway_core::{
     language::{parsed::Declaration, ty, Visibility},
 };
 use sway_types::{Ident, Spanned};
-use tower_lsp::lsp_types::{Hover, HoverContents, MarkupContent, MarkupKind, Position, Url};
+use tower_lsp::lsp_types::{self, Position, Url};
 
-pub fn hover_data(session: Arc<Session>, url: Url, position: Position) -> Option<Hover> {
-    if let Some((_, token)) = session.token_at_position(&url, position) {
+pub fn hover_data(session: Arc<Session>, url: Url, position: Position) -> Option<lsp_types::Hover> {
```

Comment:
nit: IMO, it's cleaner with `Hover` being explicitly imported from `lsp_types`

Reply:
I definitely agree with this as well but have made an exception for places that are using types from the `lsp_types` crate. 1. It makes it obvious that this is returning a type that is on the boundary between the server and the client. 2. It avoids naming conflicts as the server and the types in `lsp_types` share a lot of similar names. I noticed this is also the way rust-analyzer is working, so decided to continue with that convention.